### PR TITLE
refactor: minor rewrite to use playwright instead of puppeteer

### LINKED ISSUE #2446: Always 403 on media markt when using global.proxies or mediamarkt.proxies (but works in dotenv)
## Expected Behavior

When I put a proxy server in global.proxies or [store].proxies it should work the same as in dotenv for media markt

## Current Behavior

I hosted 10 socks5 proxies to aws and if I put them all in mediamarkt.proxies or global.proxies it immediately gives me a 403 status code.
When I put one proxy in dotenv it works.

## Steps to Reproduce

1. Host a proxy in aws
2. Put address, protocol and port in dotenv
3. Only select mediamarkt as shop
4. Start with npm run start -> everything works fine
5. Delete proxy in dotenv, make a new file called global.proxies
6. add your address from proxy there (eg socks5://myproxyUrl:myPort
7. Start script -> 403 errors on first request

## Environment

- OS:Tested on Mac or windows

## Logs

Just 403 on every request

### PR DESCRIPTION
**Pull request by klords**
### Description

Fixes #2446 #2417 #2367 potentially others

TL;DR - this fixes some issues for proxy users and provides some nice capabilities, but also requires much deeper testing to fully validate. **User beware!**

### Motivation

Several stores were no longer working with rotating proxies files. This ended up being due to the Chromium API not exposing sufficient header information to libraries brokering these requests in the proxy workflows, causing requests to be easily targeted. After a few other attempts at resolving, it became clear sidestepping Puppeteer altogether was going to be necessary due to the tight coupling with Chromium. Playwright is essentially a successor to Puppeteer (wrote by the same authors of Puppeteer after they moved to Microsoft) and maintains largely the same API as a result with a few minor differences. One of these differences, importantly, is **a first-class way to associate a proxy per page**. Huzzah.

### Caveats

As mentioned above, there are a few differences with the Playwright API:

1. **Smaller (but growing) external library/plugin support**. For example, `puppeteer-extra` is in process of being augmented so its plugins will support both Puppeteer and Playwright, but its adblocking plugin is not yet updated. As a result, the adblocking logic is replaced with essentially a ported version of the current puppeteer plugin code, and appears to work in my testing (against canyoublockit). That said, when this plugin is updated, we'll likely want to take the plugin to get further updates.
2. **Behavioral changes**. Probably the most noticeable change is that the `networkidle2` wait category has been removed in lieu of `networkidle`. This loss is causing some sites to stay open until timeout (amazon, most noticeably). There are workarounds for it (specify custom wait time added to store model, and, depending on browser selection, also blocking resources on global or per-store basis), but _**these kinds of changes mean thorough testing needs to happen before broadly releasing this**_.
3. **New features**. Speaking of browser selection, playwright ships with 3 browsers (chromium, firefox, and webkit). This seems beneficial to expose to end users as it will vary the types of browser functionality hitting sites, but of course, there are nuances to each one. Firefox is prone to network timeouts unless resources are carefully blocked, and webkit has just died on me randomly. So, while `BROWSER_TYPE` can be used to select one or the other, it's not really advertised yet

### Testing

Ran against various sites to ensure adblocking and low_bandwidth were working correctly, as well as tested various browser configurations/behaviors. This has all been very cursory so far, so user beware.

### New dependencies

`playwright`, `@cliqz/adblocker-playwright`, `playwright-extra` (not necessarily needed at this time, though I imagine the supported plugins, `recaptcha` and `humanize`, and emerging plugins will be useful to add sooner than later)

### Removed dependencies

`puppeteer`, `@doridian/puppeteer-page-proxy`, `puppeteer-extra-plugin-adblocker`, `cheerio` (this was just used to parse the DOM for the links builders when it's already been parsed in playwright... can do timing tests for a data-driven decision on whether to keep it)

### CODE REVIEW

**Comment by daningamells** [comment]
I'm just getting ERR_PROXY_CONNECTION_FAILED from using this, I am missing something here?

I need to get rid of these 403's!!!!!!!!!!!!!!!!!!!

**Comment by richie256** [comment]
I'm having net::ERR_SOCKS_CONNECTION_FAILED from using this.

**Comment by richie256** [comment]
Another Update:

I've tested it using http proxies, and I am getting STATUS CODE ERROR 407 (Proxy Authentication Required)

It's like the code doesn't use the provided username/password from the url, using that syntax: 
**http://\<username\>:\<password\>@\<ip\>:\<port\>**

as stated into the Reference guide: [https://www.jef.buzz/streetmerchant/reference/proxy/](https://www.jef.buzz/streetmerchant/reference/proxy/)

Here is the logs:
```
  pw:api => browser.newContext started +9s
  pw:api <= browser.newContext succeeded +2ms
  pw:api => browserContext.newPage started +0ms
  pw:api <= browserContext.newPage succeeded +50ms
  pw:api => page.goto started +0ms
  pw:api navigating to "about:blank", waiting until "load" +2ms
  pw:api   navigated to "about:blank" +4ms
  pw:api <= page.goto succeeded +1ms
  pw:api => page.route started +0ms
  pw:api => frame.$$eval started +0ms
  pw:api => frame.addStyleTag started +1ms
  pw:api => frame.$$eval started +0ms
  pw:api <= page.route succeeded +1ms
  pw:api => page.goto started +0ms
  pw:api navigating to "https://www.walmart.ca/en/ip/playstation5-console/6000202198562", waiting until "networkidle" +1ms
  pw:api   navigated to "https://www.walmart.ca/en/ip/playstation5-console/6000202198562" +2s
  pw:api => frame.$$eval started +1ms
  pw:api => frame.addStyleTag started +1ms
  pw:api => frame.$$eval started +1ms
  pw:api   "domcontentloaded" event fired +2ms
  pw:api   "load" event fired +0ms
  pw:api <= frame.addStyleTag failed +0ms
  pw:api <= frame.$$eval failed +1ms
  pw:api <= frame.$$eval failed +0ms
  pw:api <= frame.addStyleTag failed +4ms
  pw:api <= frame.$$eval succeeded +2ms
  pw:api <= frame.$$eval succeeded +0ms
  pw:api   "networkidle" event fired +499ms
  pw:api <= page.goto succeeded +1ms
[12 h 32 min 31 s] warn :: ✖ [1/1] [walmart-ca] [sony (sonyps5c)] ps5 console :: STATUS CODE ERROR 407
  pw:api => page.close started +1ms
  pw:api <= page.close succeeded +1ms
  pw:api => browserContext.close started +0ms
  pw:api <= browserContext.close succeeded +2ms
```
